src/libtorch.cc

```diff
@@ -1146,21 +1238,36 @@ ModelInstanceState::ProcessRequests(
   std::vector<torch::jit::IValue> input_tensors;
   bool cuda_copy = false;
   std::unique_ptr<BackendInputCollector> collector;
+  std::mutex timestamp_mu;
+
+  uint64_t compute_input_start = 0;
```

Comment:
Instead of mutex / variable combination you can use an atomic variable: std::atomic<uint64_t>

Reply:
Updated to use std::atomic<uint64_t> and removed the mutex.